mistralrs-server/src/main.rs

```diff
@@ -132,12 +133,31 @@ enum ChatCompletionResponder {
     Error(Box<dyn Error>),
 }
 
+#[derive(Serialize)]
+struct JsonError {
+    message: String,
+}
+
+impl JsonError {
+    fn new(message: String) -> Self {
+        Self { message }
+    }
+}
+
+impl IntoResponse for JsonError {
+    fn into_response(self) -> axum::response::Response {
+        let mut r = Json(self).into_response();
+        *r.status_mut() = http::StatusCode::INTERNAL_SERVER_ERROR;
+        r
+    }
+}
+
 impl IntoResponse for ChatCompletionResponder {
     fn into_response(self) -> axum::response::Response {
         match self {
             ChatCompletionResponder::Sse(s) => s.into_response(),
             ChatCompletionResponder::Json(s) => Json(s).into_response(),
-            ChatCompletionResponder::Error(e) => e.to_string().into_response(),
+            ChatCompletionResponder::Error(e) => JsonError::new(e.to_string()).into_response(),
```

Comment:
Should we split this into InternalServerError and ValidationError?

Or another mechanism to send the best HTTP status code?

Reply:
I think an InternalServerError would be best.